config/routes.rb

```diff
@@ -5,6 +5,7 @@
 
   resources :parsers do
     get :allow_flush, on: :member
+    collection { get 'datatable', constraints: { format: :json } }
```

Comment:
just my question.
Do you need `collection`?
you can simply do like this below.
From my understanding of your `spec` file.
```
get :datatable, constrains: { format: :json } }

```

Reply:
When you want to add a route inside a resource, it is a collection or a member. The syntax used in this project is different on the other routes though. I'll change it.

Source: https://guides.rubyonrails.org/routing.html#adding-member-routes

Reply:
Done :)